interfaces/builtin/polkit.go

```diff
@@ -164,19 +341,17 @@ func hasPolkitDaemonExecutable() bool {
 	return osutil.IsExecutable(polkitDaemonPath1) || osutil.IsExecutable(polkitDaemonPath2)
 }
 
-func canWriteToPolkitActionsDir() bool {
-	return unix.Access(dirs.SnapPolkitPolicyDir, unix.W_OK) == nil
-}
-
-func polkitPoliciesSupported() bool {
-	// We must have the polkit daemon present on the system and be able to write
-	// to the polkit actions directory.
-	return hasPolkitDaemonExecutable() && canWriteToPolkitActionsDir()
+func canWriteToDir(dir string) error {
+	if err := unix.Access(dir, unix.W_OK); err != nil {
+		return fmt.Errorf("%q is not writable: %w", dir, err)
+	}
+	return nil
 }
 
 func (iface *polkitInterface) StaticInfo() interfaces.StaticInfo {
 	info := iface.commonInterface.StaticInfo()
-	info.ImplicitOnCore = polkitPoliciesSupported()
+	// We must have the polkit daemon present on the system.
+	info.ImplicitOnCore = hasPolkitDaemonExecutable()
```

Comment:
this is a change of behavior vs 2.68?  which bases/since when do let write the actions dir? which bases have the daemon but don't let write?

Reply:
Good point, yes this will be (possibly unnecessary) change of behavior where the interface will show up on ubuntu-core < 24 where it can't be used since the old actions dir (`/usr/share/polkit-1/actions`) is read-only on all ubuntu-core versions (unlike the new one which is not used by the polkit interface https://github.com/canonical/core-base/blob/db6fdadeea0871a937776c4d58b4968155083f5e/static/etc/system-image/writable-paths#L74).

I think the right thing is to revert back to the original approved check, and relaxing the plug side check to only check for writable paths and not the daemon executable to match the classic slot.